### PR TITLE
Sphere Collisions and resolutions

### PR DESCRIPTION
**Pull request by BenjaFriend**
Two spheres in the debug scene that will bounce off of each other

### CODE REVIEW

File: ggp-engine/DebugScene.cpp [hunk 1]
```diff
@@ -129,8 +149,10 @@ void DebugScene::Start() {
 
 void DebugScene::Update(float _deltaTime) {
 	Scene::Update(_deltaTime);
+    // #FixForNextBuild
+    // This is rough dude
```
Comment (lampitosgames) [inline ggp-engine/DebugScene.cpp]: you're not wrong, lol

File: ggp-engine/DebugScene.cpp [hunk 2]
```diff
@@ -144,33 +166,13 @@ void DebugScene::Update(float _deltaTime) {
...
-        opposingVel.z *= -1.f;
-        entityA->ApplyForce( opposingVel );
-    }
-
-
-
-    entityA->ApplyAcceleration();*/
+    // For every physics object in the scene
+    // Update the physics
+    physicsManager->UpdatePhysics( _deltaTime );
```
Comment (lampitosgames) [inline ggp-engine/DebugScene.cpp]: This should probably be moved to the Game.cpp with the rest of the singleton update calls. We want scenes to be decoupled from singletons.

File: ggp-engine/PhysicsManager.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+PhysicsManager* PhysicsManager::Instance = nullptr;
+
+const UINT PhysicsManager::GetRigidBodyCount() const
+{
+    return rBodyCount;
+}
+
+const bool Physics::PhysicsManager::Raycast( const DirectX::XMFLOAT3 & aOrigin, const DirectX::XMFLOAT3 & aDirectoin, const float distance )
+{
+    printf( "\tStart raycast!\n" );
```
Comment (lampitosgames) [inline ggp-engine/PhysicsManager.cpp]: assuming this is going to change into real raycast code

File: ggp-engine/PhysicsManager.cpp [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    if ( Instance != nullptr )
+    {
+        delete Instance;
+        Instance = nullptr;
+    }
+}
+
+UINT PhysicsManager::AddRigidBody( Spatial* aGameObj, float aMass, EPhysicsLayer aLayer )
+{
+    assert( aGameObj );
```
Comment (lampitosgames) [inline ggp-engine/PhysicsManager.cpp]: We can use asserts, but so far I've been coding everything to fail without crashing and just output to the console. That way if there is an error, the game will still run and you can see what went wrong.

File: ggp-engine/PhysicsManager.cpp (hunk 3 above)
Comment (BenjaFriend) [reply ggp-engine/PhysicsManager.cpp]: Correct

File: ggp-engine/PhysicsManager.cpp (hunk 4 above)
Comment (BenjaFriend) [reply ggp-engine/PhysicsManager.cpp]: The asserts are removed in release mode, so its really only affecting the debugging / development cycle. We can remove them if you want to though

**Pull request merged**